common.js

```diff
@@ -74,6 +75,27 @@ function setopts (self, pattern, options) {
   self.statCache = options.statCache || Object.create(null)
   self.symlinks = options.symlinks || Object.create(null)
 
+  self.ignore = options.ignore || []
+
+  if(!Array.isArray(self.ignore)) {
+    self.ignore = [self.ignore]
+  }
+  if(self.ignore.length) {
+    self.ignore = self.ignore.map(function(pattern) {
+      var endsWithGlobStar = pattern.slice(-3) === '/**',
+          endsWithStar = pattern.slice(-2) === '/*'
+          m = { matcher: new Minimatch(pattern) };
+
+      if(endsWithGlobStar)
+        m.gmatcher = new Minimatch(pattern.slice(0, -3));
+
+      if(endsWithStar)
+        m.smatcher = new Minimatch(pattern.slice(0, -2));
```

Comment:
What if the pattern ends with multiple `/**` parts?  Eg, `foo/**/**/**` is functionally equivalent to `foo/**`